### PR TITLE
Add memory diagnostics

### PR DESCRIPTION
**Pull request by apcraig**
## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Add memory diagnostics including memory_stats namelist
- [X] Developer(s): 
    apcraig
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    Full test suite run on cheyenne, 3 compilers, all bit-for-bit
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [X] Yes
    - [ ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [X] Please provide any additional information or relevant details below:

- memory_stats default is false or off.

Unrelated to memory stats:

- cleaned up some indentation issues in ice_reprosum.F90
- removed extra box tests that seem somewhat redundant (kept east version)
- added f_narea and f_earea to ice_in

### CODE REVIEW

File: cicecore/cicedynB/infrastructure/ice_memusage_gptl.c [hunk 1]
```diff
@@ -0,0 +1,215 @@
+/*
+** $Id: get_memusage.c,v 1.10 2010-11-09 19:08:53 rosinski Exp $
+**
+** Author: Jim Rosinski
+**   Credit to Chuck Bardeen for MACOS section (__APPLE__ ifdef)
```
Comment (eclare108213) [inline cicecore/cicedynB/infrastructure/ice_memusage_gptl.c:29]: Does this code have any copyright issues?  We need to be careful with incorporating code from other sources.

**Comment by apcraig** [comment]
I've talked to Jim about this and there are no issues.  I've used it elsewhere.   It may be that we should include the COPYING file somewhere.  Let me know what you think,

https://jmrosinski.github.io/GPTL/

https://github.com/jmrosinski/GPTL/blob/master/COPYING

**Comment by apcraig** [comment]
It's failing github actions.  I was worried about that.  I'll have to figure out what to do.  Worst case, we will need a CPP.  Maybe I'll find a solution thru conda.  We'll have to decide if we want a CPP anyway, just in case there are some systems where it really won't compile.  I'll work on it.

My plan was to merge this then start testing the branch on many platforms to make sure we didn't run into any gotchas, this being one of them.

**Comment by eclare108213** [comment]
I suggest copying the COPYING file into the same module with this code and making it clear that it only applies to that.

**Comment by apcraig** [comment]
@eclare108213, Will do.  Thanks.

**Review by eclare108213 — APPROVED** [review]

**Comment by apcraig** [comment]
Just to follow up, I fixed github actions and also got things working on conda macos.  Added the copyright, and reran full test suite on cheyenne with success.  Will merge and then start testing on other platforms.

**Pull request merged**